### PR TITLE
test_timestamp_serialize: rewrite test cases in criterion

### LINKED ISSUE #2573: convert ut test_timestamp_serialize to criterion
Use Criterion to rewrite this unit test.

### PR DESCRIPTION
**Pull request by mehul-m-prajapati**
fixes #2573

### CODE REVIEW

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
do nothing -> CI won't start)

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build?
(admin: you have the next options (make sure you checked the code):
"ok to test" to accept this pull request (and further changes) for testing
"test this please" for a one time test run
do nothing -> CI won't start)

**Comment by furiel** [comment]
Just for heads up: this PR conflicts with timeutils refactor: https://github.com/balabit/syslog-ng/pull/2483.

**Comment by mehul-m-prajapati** [comment]
Hi @furiel ,

Should I merge the changes of #2483?

**Comment by furiel** [comment]
@mehul-m-prajapati Yes that would be very nice of you, thanks! 
Timeutils refactor is a very large changeset, not easy to rebase. It is also near the finish line. I hope it will be merged soon.

**Comment by Kokan** [comment]
@mehul-m-prajapati now the PR @furiel mentioned have been merged, you could rebase your work on top of that.

**Comment by mehul-m-prajapati** [comment]
@Kokan : Thanks for the update.

File: lib/logmsg/tests/test_timestamp_serialize.c [hunk 1]
```diff
@@ -22,78 +22,75 @@
...
 }
 
-static void
-test_derializing_injured_timestamp(void)
+Test(template_timestamp, test_derializing_injured_timestamp)
 {
-  PREPARE_TEST
-
-  assert_true(timestamp_serialize(sa, input_timestamps), "Failed to serialize timestamps");
+  start_grabbing_messages();
```
Comment (Kokan) [inline lib/logmsg/tests/test_timestamp_serialize.c]: Do we need to use grab message functionality here ?

Comment (mehul-m-prajapati) [reply lib/logmsg/tests/test_timestamp_serialize.c]: @Kokan : I can see this functionality in the previous version of code.

Comment (mehul-m-prajapati) [reply lib/logmsg/tests/test_timestamp_serialize.c]: Yes, you are correct.

**Comment by MrAnno** [comment]
@kira-syslogng test this please

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**